### PR TITLE
Make rshift implementation consistent between pytorch and numpy

### PR DESCRIPTION
**Pull request by ifedan**
Currently rshift function is inconsistent between numpy and pytorch:
```
>>> import torch
>>> import numpy as np
>>> a = torch.tensor([-1], dtype=torch.int, device="cpu")
>>> a >> 1
tensor([2147483647], dtype=torch.int32)
>>> a = torch.tensor([-1], dtype=torch.int, device="cuda")
>>> a >> 1
tensor([-1], device='cuda:0', dtype=torch.int32)
>>> a = np.array([-1])
>>> a >> 1
array([-1])
```

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -1479,6 +1479,26 @@ def test_all_any_empty(self):
         self.assertTrue(x.all())
         self.assertFalse(x.any())
 
+    def test_rshift_same_as_numpy(self):
```
Comment (gchanan) [inline test/test_torch.py]: this needs to check that numpy is available (see other tests)

**Comment by apaszke** [comment]
Or maybe we should just forbid shifts for signed types? Aren’t those implementation-defined in C++?

**Review by apaszke — CHANGES_REQUESTED** [review]
Yep, shifts of signed numbers are implementation-dependent, and shifts of floating points are not supported in C/C++. I really think we should forbid those operations for all but unsigned types.

**Comment by gchanan** [comment]
@apaszke I would agree if we had good support unsigned int types, so users could easily specify what they wanted, but we don't.  There's just no good alternative.

Given that, plus these other reasons below, I'm inclined to accept this:
1) we already supplied this functionality to users (if inconsistently)
2) numpy implements this the same way
3) we are literally just translating the C++ call (I'm okay putting a warning though).

File: aten/src/TH/generic/THTensorEvenMoreMath.cpp [hunk 2]
```diff
@@ -831,15 +831,15 @@ void THTensor_(rshift)(THTensor *r_, THTensor *t, scalar_t value)
 #if defined(TH_REAL_IS_BYTE)
         rp[i] = ((scalar_t) tp[i]) >> value;
 #else
-        rp[i] = ((ureal) tp[i]) >> value;
+        rp[i] = ((accreal) tp[i]) >> value;
```
Comment (gchanan) [inline aten/src/TH/generic/THTensorEvenMoreMath.cpp]: why is doing this in terms of accreal the correct thing to do as opposed to scalar_t?

**Review by gchanan — CHANGES_REQUESTED** [review]
Please provide a better PR description; there's no explanation of the issue (i.e. discussion about how it's implementation defined, what the BC-breaking change is).

**Comment by apaszke** [comment]
I don't know, shifts on signed types feel like something you _really_ don't want to do... I would really be ok and would feel much better if we limited this only to byte tensors.

**Pull request closed without merging**